### PR TITLE
feat(reactjs-jsx-ppx): react/jsx-runtime transform

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

File: reactjs-jsx-ppx/reactjs_jsx_ppx_v3.ml [hunk 1]
```diff
@@ -540,9 +538,8 @@ let jsxMapper =
                   Ocaml_common.Location.prerr_warning pattern.ppat_loc
                     (Preprocessor
                        (Printf.sprintf
-                          "ReasonReact: optional argument annotations must \
-                           have explicit `option`. Did you mean \
-                           `option(%s)=?`?"
+                          "React: optional argument annotations must have \
```
Comment (jchavarri) [inline reactjs-jsx-ppx/reactjs_jsx_ppx_v3.ml:541]: Maybe we could refer to the ppx name to make it clearer where the error is coming from, as there are a few parts involved when using React components in Reason:
- the ReactJS bindings
- the Reason parser JSX transformation
- and the reactjs-jsx-ppx preprocessor

e.g.

```suggestion
                          "reactjs-jsx-ppx: optional argument annotations must have \
```

File: reactjs-jsx-ppx/reactjs_jsx_ppx_v3.ml [hunk 2]
```diff
@@ -374,37 +427,25 @@ let makeExternalDecl fnName loc namedArgListWithKeyAndRef namedTypeList =
 
 (* TODO: some line number might still be wrong *)
 let jsxMapper =
-  let jsxVersion = ref None in
```
Comment (jchavarri) [inline reactjs-jsx-ppx/reactjs_jsx_ppx_v3.ml:377]: Was this change required for the jsx transform? Or just cleanup?

File: reactjs-jsx-ppx/reactjs_jsx_ppx_v3.ml [hunk 3]
```diff
@@ -428,83 +469,40 @@ let jsxMapper =
...
-    (* React.createElement(Component.make, props, ...children) *)
-    match !childrenArg with
-    | None ->
-        Exp.apply ~loc ~attrs
-          (Exp.ident ~loc { loc; txt = Ldot (Lident "React", "createElement") })
-          [ (nolabel, Exp.ident ~loc { txt = ident; loc }); (nolabel, props) ]
-    | Some children ->
-        Exp.apply ~loc ~attrs
-          (Exp.ident ~loc
-             { loc; txt = Ldot (Lident "React", "createElementVariadic") })
```
Comment (jchavarri) [inline reactjs-jsx-ppx/reactjs_jsx_ppx_v3.ml:443]: Same as with `createDOMElementVariadic `. This function is used in a few tests that I believe can be updated.

**Review by jchavarri — APPROVED** [review]
Looks great! What would be the release plan? I understand it relies on both the reason-react changes, and the ReactJS package being > 17 right? or 18?

**Comment by anmonteiro** [comment]
> What would be the release plan? 

I think we can probably get away with:
- releasing the ppx to opam as v0.1
- merging this pr
- release as v1.0


Then  consumers would be able to select which version of the ppx they want in their projects.

File: reactjs-jsx-ppx/reactjs_jsx_ppx_v3.ml (hunk 2 above)
Comment (anmonteiro) [reply reactjs-jsx-ppx/reactjs_jsx_ppx_v3.ml:377]: just cleanup!

**Comment by anmonteiro** [comment]
tracking the rest of this in https://github.com/reasonml/reason-react/pull/714
